From 989461bc91fca3bdf169d779c5b3bc7cb379f9a8 Mon Sep 17 00:00:00 2001
From: Gu Zitao <guzitao@wxiat.com>
Date: Wed, 16 Nov 2022 09:16:10 +0800
Subject: [PATCH] sw64: fix kernel_stack_pointer
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 989461bc91fca3bdf169d779c5b3bc7cb379f9a8
Modified-by-SEL: No


Sunway inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I645TO

--------------------------------

Signed-off-by: Gu Zitao <guzitao@wxiat.com>
Reviewed-by: He Sheng <hesheng@wxiat.com>
Signed-off-by: Gu Zitao <guzitao@wxiat.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/sw_64/include/asm/ptrace.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/sw_64/include/asm/ptrace.h b/arch/sw_64/include/asm/ptrace.h
index efc93e731b2a..4db8b61fc093 100644
--- a/arch/sw_64/include/asm/ptrace.h
+++ b/arch/sw_64/include/asm/ptrace.h
@@ -55,7 +55,7 @@ struct pt_regs {
 #define profile_pc(regs) instruction_pointer(regs)
 #define current_user_stack_pointer() rdusp()
 #define user_stack_pointer(regs) rdusp()
-#define kernel_stack_pointer(regs) (((regs->ps) >> 4) & (TASK_SIZE - 1))
+#define kernel_stack_pointer(regs) ((unsigned long)((regs) + 1))
 #define instruction_pointer_set(regs, val) ((regs)->pc = val)
 
 
-- 
2.33.0

